java/src/com/google/idea/blaze/java/run/hotswap/BlazeReloadFileAction.java

```diff
@@ -194,16 +223,38 @@ private void findAndCopyOutputFile(VirtualFile sourceFile, File tempClassDir, St
     @Override
     public void update(@NotNull AnActionEvent e) {
         delegate.update(e);
-        if (e.getPresentation().isEnabled()) {
-            DebuggerSession session =
-                    DebuggerManagerEx.getInstanceEx(e.getProject()).getContext().getDebuggerSession();
-            VirtualFile vf = e.getData(CommonDataKeys.VIRTUAL_FILE);
-            if (session != null && vf != null) {
-                VirtualFile sourceRoot = ProjectFileIndex.getInstance(e.getProject()).getSourceRootForFile(vf);
-                e.getPresentation().setEnabled(sourceRoot != null);
+        // We want to give a hint why action is not enabled so we need to have it visible at all times this is possible
```

Comment:
These changes seem a bit much, especially replacing the action text

Reply:
This is based on the amount of confusion that users have so far. Feel free to drop this from your PR after the merge.